crates/cheatcodes/src/crypto.rs

```diff
@@ -89,14 +91,56 @@ impl Cheatcode for rememberKeyCall {
     fn apply(&self, state: &mut Cheatcodes) -> Result {
         let Self { privateKey } = self;
         let wallet = parse_wallet(privateKey)?;
-        let address = wallet.address();
-        if let Some(script_wallets) = state.script_wallets() {
-            script_wallets.add_local_signer(wallet);
-        }
+        let address = inject_wallet(state, wallet);
         Ok(address.abi_encode())
     }
 }
 
+impl Cheatcode for rememberKeys_0Call {
+    fn apply(&self, state: &mut Cheatcodes) -> Result {
+        let Self { mnemonic, derivationPath, count } = self;
+        tracing::info!("Remembering {} keys", count);
+        let wallets = derive_wallets::<English>(mnemonic, derivationPath, *count)?;
+
+        tracing::info!("Adding {} keys to script wallets", count);
+
+        let mut addresses = Vec::<Address>::with_capacity(wallets.len());
+        for wallet in wallets {
+            let addr = inject_wallet(state, wallet);
+            addresses.push(addr);
+        }
+
+        Ok(addresses.abi_encode())
+    }
+}
+
+impl Cheatcode for rememberKeys_1Call {
+    fn apply(&self, state: &mut Cheatcodes) -> Result {
+        let Self { mnemonic, derivationPath, language, count } = self;
+        let wallets = derive_wallets_str(mnemonic, derivationPath, language, *count)?;
+        let mut addresses = Vec::<Address>::with_capacity(wallets.len());
+        for wallet in wallets {
+            let addr = inject_wallet(state, wallet);
+            addresses.push(addr);
+        }
+
+        Ok(addresses.abi_encode())
+    }
+}
+
+fn inject_wallet(state: &mut Cheatcodes, wallet: LocalSigner<SigningKey>) -> Address {
+    let address = wallet.address();
+    if let Some(script_wallets) = state.script_wallets() {
+        script_wallets.add_local_signer(wallet);
+    } else {
+        // This is needed in case of testing scripts, wherein script wallets are not set on setup.
+        let script_wallets = ScriptWallets::new(MultiWallet::default(), None);
+        script_wallets.add_local_signer(wallet);
+        Arc::make_mut(&mut state.config).script_wallets = Some(script_wallets);
```

Comment:
this might be quite expensive because we'll clone config which stores all project bytecodes

I think essentially for tests `ScriptWallets` can just be a vec of `PrivateKeySigner`s stored on `Cheatcodes`